src/components/FieldViolationMessages.tsx

```diff
@@ -0,0 +1,26 @@
+import React, {useMemo} from 'react';
+import {View} from 'react-native';
+import useLocalize from '@hooks/useLocalize';
+import ViolationsUtils from '@libs/Violations/ViolationsUtils';
+import useThemeStyles from '@styles/useThemeStyles';
+import {TransactionViolation} from '@src/types/onyx';
+import Text from './Text';
+
+export default function FieldViolationMessages({violations}: {violations: TransactionViolation[]}) {
+    const styles = useThemeStyles();
+
+    const localize = useLocalize();
+    const translate = localize.translate;
+
+    const violationMessages = useMemo(() => violations.map((violation) => [violation.name, ViolationsUtils.getViolationTranslation(violation, translate)]), [translate, violations]);
+
+    return (
+        <>
+            {violationMessages.map(([name, message]) => (
+                <View key={`violationMessages.${name}`}>
```

Comment:
Out of curiosity, can we use the `message` in the key instead of the `name`? If so, let's simplify this and only pass `name`, and return an array of names in the useMemo function, instead of an array of `name, message`

Reply:
No we can't use the message as the key. This is the translation key that we use to get the message.

Reply:
The `name` is passed as the key prop for the message views, so it needs to be unique.

We know that each violation will only appear once, but there are some violations that could present the same message to the user:

```ts
overLimit: ({amount}: ViolationsOverLimitParams) => `Amount over ${amount}/person limit`,
overLimitAttendee: ({amount}: ViolationsOverLimitParams) => `Amount over ${amount}/person limit`,
```

So we need something to distinguish them to ensure uniqueness.

Reply:
I can probably get away with using the array index to distinguish them, since we're confident they won't be changing frequently.

Reply:
Ah no, they are one step ahead of us:
![CleanShot 2023-12-12 at 11 58 10@2x](https://github.com/Expensify/App/assets/22041394/7e9a04db-72a4-445f-a41d-4665f3da0a15)